simulator/src/main.rs

```diff
@@ -133,20 +137,64 @@ fn main() -> std::io::Result<()> {
     Ok(())
 }
 
-struct OpensslCrypto;
+struct SimCrypto;
 
-impl Crypto for OpensslCrypto {
+impl Crypto for SimCrypto {
+    type Cdi = Vec<u8>;
+
+    /// Uses incrementing values for each byte to ensure tests are
+    /// deterministic
     fn rand_bytes(dst: &mut [u8]) -> Result<(), DpeErrorCode> {
-        openssl::rand::rand_bytes(dst).map_err(|_| DpeErrorCode::InternalError)
+        OpensslCrypto::rand_bytes(dst).map_err(|_| DpeErrorCode::InternalError)
     }
 
-    fn _hash(profile: DpeProfile, bytes: &[u8], digest: &mut [u8]) -> Result<(), DpeErrorCode> {
-        use openssl::hash::{hash, MessageDigest};
-        let alg = match profile {
+    fn hash(profile: DpeProfile, bytes: &[u8], digest: &mut [u8]) -> Result<(), DpeErrorCode> {
+        let md = match profile {
             DpeProfile::P256Sha256 => MessageDigest::sha256(),
             DpeProfile::P384Sha384 => MessageDigest::sha384(),
         };
-        digest.copy_from_slice(&hash(alg, bytes).map_err(|_| DpeErrorCode::InternalError)?);
-        Ok(())
+
+        OpensslCrypto::hash(bytes, digest, md).map_err(|_| DpeErrorCode::InternalError)
+    }
+
+    fn derive_cdi(
+        profile: DpeProfile,
+        measurement_digest: &[u8],
+        info: &[u8],
+    ) -> Result<Self::Cdi, DpeErrorCode> {
+        let md = match profile {
+            DpeProfile::P256Sha256 => MessageDigest::sha256(),
+            DpeProfile::P384Sha384 => MessageDigest::sha384(),
+        };
```

Comment:
It looks like we could do a helper function here. Do you think its worth it?

```Rust
impl SimCrypto {
    fn get_digest(profile: DpeProfile) -> MessageDigest {
        match profile {
            DpeProfile::P256Sha256 => MessageDigest::sha256(),
            DpeProfile::P384Sha384 => MessageDigest::sha384(),
        }
    }
}
```

Reply:
Good idea. Done.